File: src/main/kotlin/com/developersam/friend/FriendData.kt
package com.developersam.friend

import com.developersam.auth.GoogleUser

/**
 * [FriendData] is a combination of friend [list] and [requests] for the given user.
 */
class FriendData private constructor(
        private val list: List<GoogleUser>, private val requests: List<GoogleUser>
) {

    /**
     * Construct the [FriendData] for the given [user].
     */
    constructor(user: GoogleUser): this(list = FriendPair[user], requests = FriendRequest[user])

}

File: frontend/src/app/friends/friend-data.ts
import { GoogleUser } from '../shared/google-user';

/**
 * [FriendData] is the collection of friend related data].
 */
export interface FriendData {
  /**
   * A list of friends.
   */
  list: GoogleUser[];
  /**
   * A list of friend requests.
   */
  requests: GoogleUser[];
}
